core/store/src/lib.rs

```diff
@@ -459,7 +459,7 @@ impl Store {
 /// Keeps track of current changes to the database and can commit all of them to the database.
 pub struct StoreUpdate {
     transaction: DBTransaction,
-    storage: Arc<dyn Database>,
+    store: Store,
```

Comment:
I have no clue what sort of abstraction layers are we breaking here. I guess if it compiles and all tests pass it should be fine. 🙏  In compiler we trust. 🙏